slm_porosity/heat_transfer.f90

```diff
@@ -737,11 +662,29 @@ FUNCTION porosity_term ()
     porosity_term = (1.0_pr - u(:,n_var_porosity)) / (1.0_pr - initial_porosity)
   END FUNCTION porosity_term
 
-  PURE FUNCTION conductivity (temperature)
+  FUNCTION conductivity (temperature)
     IMPLICIT NONE
     REAL (pr), INTENT(IN) :: temperature(:)
     REAL (pr) :: conductivity(SIZE(temperature))
-    conductivity = 1.0_pr + conductivity_der*temperature
+    REAL (pr) :: temperature_S, temperature_L
+    REAL (pr) :: enthalpy_L, enthalpy_S
+    LOGICAL :: is_D   !is_D = 0 cause there is now changes in DRHS for now
+    is_D = 0
+    enthalpy_S = 1.0_pr - fusion_delta/2                ! enthalpy at the solidus temperature
+    enthalpy_L = 1.0_pr + fusion_delta/2 + fusion_heat  ! enthalpy at the liquidus temperature
+    temperature_L = enthalpy_L - fusion_heat
```

Comment:
There wouldn't be any changes because enthalpy_S|L must be in input file. As for me normalizing mean value on one could be temporary. Also firstly it was temperature_S|L = temperature(enthalpy_S|L) but, as discussed, this function didn't work well, so this is also a temporary solution.